### PR TITLE
[tensorboard] add tests for add_custom_scalars and others

### PR DESCRIPTION
**Pull request by lanpa**
Originally, the tests for tensorboard writer are smoke tests only. This PR lets CI compare the output with expected results at low level. The randomness of the tensors in the test are also removed.
ps. I found that how protobuf serializes data differs between different python environment. One method to solve this is to write the data and then read it back instantly. (compare the data at a higher level)

For `add_custom_scalars`, the data to be written is a dictionary. and the serialized result might be different (not `ordereddict`). So only smoke test for that.

### CODE REVIEW

File: test/expect/caffe_mnist.expect [hunk 1]
```diff
@@ -0,0 +1,892 @@
+node {
+  name: "conv1/XavierFill"
```
Comment (orionr) [inline test/expect/caffe_mnist.expect]: Do you still need this file? If so, can you rename to `test/expect/TestTensorBoard.test_mnist.expect` or something?

File: test/test_tensorboard.py [hunk 2]
```diff
@@ -348,56 +352,57 @@ def test_video(self):
...
 
         module_id = function_ptr.__class__.__module__
+        test_dir = os.path.dirname(sys.modules[module_id].__file__)
         functionName = function_ptr.id().split('.')[-1]
-        test_file = os.path.realpath(sys.modules[module_id].__file__)
-        expected_file = os.path.join(os.path.dirname(test_file),
+        expected_file = os.path.join(test_dir,
                                      "expect",
-                                     module_id.split('.')[-1] + '.' + functionName + ".expect")
+                                     'TestTensorboard.' + functionName + ".expect")
```
Comment (orionr) [inline test/test_tensorboard.py]: Nit - can we change `TestTensorboard.*.expect` to `TestTensorBoard.*.expect` with the B capitalized?

File: test/test_tensorboard.py [hunk 3]
```diff
@@ -348,56 +352,57 @@ def test_video(self):
...
     def write_proto(str_to_compare, function_ptr):
         module_id = function_ptr.__class__.__module__
+        test_dir = os.path.dirname(sys.modules[module_id].__file__)
         functionName = function_ptr.id().split('.')[-1]
-        test_file = os.path.realpath(sys.modules[module_id].__file__)
-        expected_file = os.path.join(os.path.dirname(test_file),
+        expected_file = os.path.join(test_dir,
                                      "expect",
-                                     module_id.split('.')[-1] + '.' + functionName + ".expect")
+                                     'TestTensorboard.' + functionName + ".expect")
```
Comment (orionr) [inline test/test_tensorboard.py]: Same here?

**Review by orionr — CHANGES_REQUESTED** [review]
A few nits. Thanks!

**Comment by orionr** [comment]
Also, can you update the description? Thanks.

File: test/expect/caffe_mnist.expect (hunk 1 above)
Comment (lanpa) [reply test/expect/caffe_mnist.expect]: Let's make it a separate PR.

**Review by orionr — APPROVED** [review]
Looks great! Waiting for tests to pass, but importing to run internal tests too.

**Comment by orionr** [comment]
Looks like we got rate limited by GitHub on the tests again. :( cc @ezyang. All passed, though, so landing internally.

**Pull request closed without merging**